collectors/timex.plugin/plugin_timex.c

```diff
@@ -44,8 +44,12 @@ void *timex_main(void *ptr)
 
         struct timex timex_buf = {};
         int sync_state = 0;
-        sync_state = adjtimex(&timex_buf);
 
+#if defined(__FreeBSD__) || defined(__APPLE__)
+        sync_state = ntp_adjtime(&timex_buf);
+#else
+        sync_state = adjtimex(&timex_buf);
+#endif
```

Comment:
Please, move this code under a new function in `libnetdata/os.c` (something like `adjust_time()` or something similar).

We want to reduce the amount of OS-specific macros outside of `libnetdata`.

Reply:
good point..I'll make changes

Reply:
I think aliasing something like below in os.h should be ok in this use case.
`#include <sys/timex.h>
#if defined(__FreeBSD__) || defined(__APPLE__)
#define ADJUST_TIMEX(x) ntp_adjtime(x)
#else
#define ADJUST_TIMEX(x) adjtimex(x)
#endif`

Reply:
In my opinion, it's better to use CAPS with macros so as to distinguish them with regular variables/functions

Reply:
> In my opinion, it's better to use CAPS with macros so as to distinguish them with regular variables/functions

Agreed, but the rest of the code already uses lowercase.

Reply:
> Agreed, but the rest of the code already uses lowercase.

Not everywhere. Even in this file, we use uppercase.

Reply:
this is partly true, I have seen both lowercase and uppercase macros in our code. But if you see the os.h itself where I've made changes, other macros are in uppercase. I still think, it's better to go with uppercase macros.

Reply:
> > Agreed, but the rest of the code already uses lowercase.
> 
> Not everywhere. Even in this file, we use uppercase.

...for function-like macros.

We are debating a trivial thing here. For the time being, this will be used only by timex. Let's pick the approach you prefer and merge the PR.

Reply:
yeah, this is really trivial thing. lets merge this PR for now.